ray-operator/test/sampleyaml/raycluster_test.go

```diff
@@ -13,6 +13,10 @@ import (
 )
 
 func TestRayCluster(t *testing.T) {
+	sema := make(chan struct{}, 3)
+	for i := 0; i < cap(sema); i++ {
+		sema <- struct{}{}
+	}
```

Comment:
This e2e test now takes too long (~30min) and timeouts the CI. So we parallelize the sub-tests with a semaphore to allow 3 parallelism.